api/src/controllers/login.js

```diff
@@ -355,6 +456,49 @@ module.exports = {
       });
   },
 
+  getPasswordReset: (req, res, next) => {
+    return renderPasswordReset(req)
+      .then(body => {
+        res.setHeader(
+          'Link',
+          '</login/style.css>; rel=preload; as=style, '
+              + '</login/auth-utils.js>; rel=preload; as=script, '
+              + '</login/password-reset.js>; rel=preload; as=script'
+        );
+        res.send(body);
+      })
+      .catch(next);
+  },
+  resetPassword: async (req, res) => {
+    const limited = await rateLimitService.isLimited(req);
+    if (limited) {
+      return serverUtils.rateLimited(req, res);
+    }
+
+    try {
+      const validation = validatePasswordReset(req.body.password, req.body.confirmPassword);
+      if (!validation.isValid) {
+        return res.status(400).json({
+          error: validation.error,
+          params: validation.params,
+        });
+      }
+      const user = await db.users.get(`org.couchdb.user:${req.body.user}`);
+      await updatePassword(user, req.body.password);
```

Comment:
I think there's a big security hole here! What if I navigate to the reset password page, set my userCtx cookie to `{ name: ben }`, submit a new password. I haven't tried but I _think_ that will update YOUR password to whatever I want.

I think the submission of the password reset form MUST contain the existing password as well as the new password otherwise how can you verify the user is who they claim to be?